src/julia.h

```diff
@@ -179,6 +179,20 @@ STATIC_INLINE int jl_array_ndimwords(uint32_t ndims)
     return (ndims < 3 ? 0 : ndims-2);
 }
 
+typedef struct {
+    union {
```

Comment:
add `JL_DATA_TYPE` here to hint to the reader that this is a subtype of `jl_value_t`